keel-core/src/main/kotlin/com/netflix/spinnaker/keel/services/ApplicationService.kt

```diff
@@ -611,3 +650,15 @@ fun DeliveryConfig.contexts(
   versions.flatMap { version ->
     environments.map { env -> VerificationContext(this, env, version) }
   }
+
+/**
+ * Holds the info we need about artifacts in an environment for building the UI view.
+ *
+ * This is used in a list of versions pertaining to a specific delivery artifact.
+ */
+data class StatusInfoForArtifactInEnvironment(
```

Comment:
Should this class maybe be defined in `keel-sql` instead, since that's where the data is coming from and `keel-core` already has a dependency on it?

Reply:
🤷 I like defining it here because it's only used here. I think it really matters.